File: QE-Java Selenium/Activity13.py
from selenium import webdriver
from selenium.webdriver.common.by import By

driver = webdriver.Firefox()

driver.get("https://training-support.net/webelements/tables")
driver.maximize_window()

print("Page Title:", driver.title)

table_xpath = "/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr"

rows = driver.find_elements(By.XPATH, table_xpath)
row_count = len(rows)

columns = driver.find_elements(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr[1]/td")
column_count = len(columns)

print("Number of Rows:", row_count)
print("Number of Columns:", column_count)

print("Third Row Values:")

third_row = driver.find_elements(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr[3]/td")

for cell in third_row:
    print(cell.text)

cell_2_2 = driver.find_element(By.XPATH,"/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr[2]/td[2]")
print("2nd Row 2nd Column:", cell_2_2.text)

driver.quit()

File: QE-Java Selenium/Activity13.java
package demos;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;

import java.util.List;

public class Activity13 {

    public static void main(String[] args) {

        WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/tables");
        driver.manage().window().maximize();

        System.out.println("Page Title: " + driver.getTitle());

        List<WebElement> rows = driver.findElements(
                By.xpath("/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr")
        );
        System.out.println("Number of Rows: " + rows.size());

        List<WebElement> columns = driver.findElements(
                By.xpath("/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr[1]/td")
        );
        System.out.println("Number of Columns: " + columns.size());

        System.out.println("Third Row Values:");

        List<WebElement> thirdRowCells = driver.findElements(
                By.xpath("/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr[3]/td")
        );

        for (WebElement cell : thirdRowCells) {
            System.out.println(cell.getText());
        }

        WebElement cell_2_2 = driver.findElement(
                By.xpath("/html/body/div/main/div/div/div/div/div[2]/table/tbody/tr[2]/td[2]")
        );
        System.out.println("2nd Row 2nd Column: " + cell_2_2.getText());

        driver.quit();
    }
}

